src/main/resources/pip-inspector.py

```diff
@@ -128,43 +127,65 @@ def populate_dependency_tree(project_root_node, requirements_path):
 
 def recursively_resolve_dependencies(package_name, history):
     """Forms a DependencyNode by recursively resolving its dependencies. Tracks history for cyclic dependencies."""
-    package = get_package_by_name(package_name)
+    dependency_node, child_names = get_package_by_name(package_name)
 
-    if package is None:
+    if dependency_node is None:
         return None
 
-    dependency_node = DependencyNode(package.project_name, package.version)
-
-    if package_name.lower() not in history:
-        history.append(package_name.lower())
-        for package_dependency in package.requires():
-            child_node = recursively_resolve_dependencies(package_dependency.key, history)
+    if dependency_node.name not in history:
+        for child_name in child_names:
+            child_node = recursively_resolve_dependencies(child_name, history + [dependency_node.name])
```

Comment:
Previously we were modifying the history list and then passing the modified value here. Now we pass in updated history but keep the local variable as-is. This preserves the handling of potential cyclical dependencies but also ensures that the same transitive dependency can be shown in different parts of the dependency tree.